lib/shared/input_dialogs.dart

```diff
@@ -167,6 +169,76 @@ Widget buildCommunitySuggestionWidget(payload, {void Function(CommunityView)? on
   );
 }
 
+/// Shows a dialog which allows typing/search for an instance
+void showInstanceInputDialog(BuildContext context, {required String title, required void Function(Instance) onInstanceSelected, Iterable<Instance>? emptySuggestions}) async {
+  Future<String?> onSubmitted({Instance? payload, String? value}) async {
+    if (payload == null) return null;
+
+    onInstanceSelected(payload);
+    Navigator.of(context).pop();
+    return null;

```

Comment:
See my overall comments about this.

Reply:
Addressed in 88b50b7ea453b38fc548ed2fab1f440ee5c8417a